cranelift-codegen/src/verifier/mod.rs

```diff
@@ -151,6 +111,50 @@ fn format_context(context: &Option<String>) -> String {
     }
 }
 
+/// Convenience converter for making error-reporting less verbose.
+///
+/// Converts a tuple of `(location, context, message)` to a `VerifierError`.
+/// ```
+/// use cranelift_codegen::verifier::VerifierErrors;
+/// use cranelift_codegen::ir::Inst;
+/// let mut errors = VerifierErrors::new();
+/// errors.report((Inst::from_u32(42), "v3 = iadd v1, v2", "iadd cannot be used with values of this type"));
+/// // note the double parenthenses to use this syntax
+/// ```
+impl<L, C, M> From<(L, C, M)> for VerifierError
+where
+    L: Into<AnyEntity>,
+    C: Into<String>,
+    M: Into<String>,
+{
+    fn from(items: (L, C, M)) -> Self {
+        let (location, context, message) = items;
+        Self {
+            location: location.into(),
+            context: Some(context.into()),
+            message: message.into(),
+        }
+    }
+}
+
+/// Convenience converter for making error-reporting less verbose.
+///
+/// Same as above but without `context`.
+impl<L, M> From<(L, M)> for VerifierError
+where
+    L: Into<AnyEntity>,
+    M: Into<String>,
+{
+    fn from(items: (L, M)) -> Self {
+        let (location, message) = items;
+        Self {
+            location: location.into(),
+            context: None,
+            message: message.into(),
+        }
+    }
+}
+
```

Comment:
These convenience implementations make it possible to be a bit less verbose... I was trying to make this as easy-to-use and concise as the macros and this was as close as I could get.